editor.planx.uk/src/pages/FlowEditor/lib/store/navigation.ts

```diff
@@ -115,21 +118,39 @@ export const navigationStore: StateCreator<
   },
 
   /**
-   * Calculate the status of each section node based on the current position in a flow
+   * Calculate the status of each section node based on the current position in a flow (eg state.cachedBreadcrumbs) or provided breadcrumbs (eg on reconciliation)
    *   ** will need to become aware of all breadcrumbs within a section in the near future for "back" rather than only currentCard & upcomingCardIds
    */
-  sectionStatuses: (): Record<string, SectionStatus> => {
-    const { sectionNodes, currentCard, upcomingCardIds, cachedBreadcrumbs } =
-      get();
+  sectionStatuses: (
+    breadcrumbs?: Store.breadcrumbs,
+    updatedNodeIds?: string[]
+  ): Record<string, SectionStatus> => {
```

Comment:
this is a big change: `sectionStatuses()` now accepts optional arguments - if provided, it will calculate the section status based off the breadcrumbs object provided. if no arguments are provided, it will calculate the section status based off of cached breadcrumbs in state just as before.

this let's us calculate section statuses on the reconciliation page using the breadcrumbs object returned by /validate-session endpoint that is _not_ in state yet. 

open to other approaches for accomplishing this, but this felt most straightforward & flexible for now!